File: demo/src/main/java/com/example/demo/entity/Registration.java
package com.example.demo.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Registration {
    private Long id;
    private Long patientId;
    private Long doctorId;
    private String patientName;
    private String doctorName;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime appointmentTime;
    private String status;
    private String notes;
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
}

File: frontend/src/api/modules/registration.ts
import request from '../request';

export interface Registration {
  id?: number;
  patientId: number;
  doctorId: number;
  patientName?: string;
  doctorName?: string;
  appointmentTime: string;
  status?: string;
  notes?: string;
  createdAt?: string;
}

export const fetchRegistrations = (params?: {
  patientName?: string;
  doctorName?: string;
  status?: string;
}) => request.get<Registration[]>('/registrations', { params });

export const createRegistration = (registration: Registration) => request.post('/registrations', registration);

export const updateRegistration = (id: number, registration: Registration) =>
  request.put(`/registrations/${id}`, registration);

export const removeRegistration = (id: number) => request.delete(`/registrations/${id}`);

